### PR TITLE
Support for Intel QuickAssist ECC KeyGen acceleration

### PR DESCRIPTION
**Pull request by dgarske**
# Description

Support for Intel QuickAssist ECC KeyGen acceleration

# Testing

```
./configure --enable-sniffer --enable-asynccrypt CFLAGS="-DWOLFSSL_SNIFFER_CHAIN_INPUT" --enable-curve25519 --enable-curve448 --enable-session-ticket --with-intelqa=../QAT1.8 && make
```

# Checklist

 - [ ] added tests
 - [ ] updated/added doxygen
 - [ ] updated appropriate READMEs
 - [ ] Updated manual and documentation

### CODE REVIEW

File: wolfcrypt/src/ecc.c [hunk 1]
```diff
@@ -4780,24 +4787,23 @@ static WC_INLINE void wc_ecc_reset(ecc_key* key)
  * Note this function is local to the file because of the argument type
  *      ecc_curve_spec. Having this argument allows for not having to load the
  *      curve type multiple times when generating a key with wc_ecc_make_key().
+ * For async the results are placed directly into pubOut, so this function 
+ *      does not need called again
```
Comment (SparkiDev) [inline wolfcrypt/src/ecc.c:4791]: "does not need to be called again"

File: wolfcrypt/src/ecc.c [hunk 2]
```diff
@@ -4780,24 +4787,23 @@ static WC_INLINE void wc_ecc_reset(ecc_key* key)
...
         ecc_point* pubOut, WC_RNG* rng)
 {
     int err = MP_OKAY;
-#if !defined(WOLFSSL_ATECC508A) && !defined(WOLFSSL_ATECC608A) \
-  && !defined(WOLFSSL_SILABS_SE_ACCEL) && !defined(WOLFSSL_KCAPI_ECC)
-#if !defined(WOLFSSL_SP_MATH)
+#ifdef HAVE_ECC_MAKE_PUB
+#if !defined(WOLFSSL_SP_MATH) && !(defined(WOLFSSL_ASYNC_CRYPT) && \
+        defined(WC_ASYNC_ENABLE_ECC) && defined(HAVE_INTEL_QA))
     ecc_point* base = NULL;
```
Comment (SparkiDev) [inline wolfcrypt/src/ecc.c:4802]: Scope of base can be reduced too.

File: wolfcrypt/src/ecc.c [hunk 3]
```diff
@@ -4855,10 +4858,26 @@ static int ecc_make_pub_ex(ecc_key* key, ecc_curve_spec* curveIn,
...
 
+#if defined(WOLFSSL_ASYNC_CRYPT) && defined(WC_ASYNC_ENABLE_ECC) && \
+    defined(HAVE_INTEL_QA)
+    if (err == MP_OKAY) {
+        word32 keySz = key->dp->size;
+        /* sync private key to raw */
+        err = wc_mp_to_bigint_sz(&key->k, &key->k.raw, keySz);
+        if (err == MP_OKAY) {
+            err = IntelQaEccPointMul(&key->asyncDev,
+                &key->k.raw, pub->x, pub->y, pub->z,
```
Comment (SparkiDev) [inline wolfcrypt/src/ecc.c:4869]: Should this be pub->x->raw?

**Review by SparkiDev — CHANGES_REQUESTED** [review]

File: wolfcrypt/src/ecc.c (hunk 3 above)
Comment (dgarske) [reply wolfcrypt/src/ecc.c:4865]: For the ECC key gen I am passing in the mp_int and populating directly in the Intel QAT callback. The NUMA raw is allocated in the quickassist.c instead. This allowed implementation without having to setup states in key gen and make pub.

**Review by SparkiDev — APPROVED** [review]

**Pull request merged**